### PR TITLE
chore(eslint): enable eslint for ts (WIP)

### PR DESCRIPTION
**Pull request by rhelling**
- [x] Update settings to current typescript standard
- [x] Fix ts issues 
- [x] Re-enable eslint for js
- [x] Fix js issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,11 +26,10 @@
         "unit:debug": "cross-env BABEL_ENV=test karma start test/karma.conf.js --debug",
         "snapshots": "cross-env BABEL_ENV=jest jest --silent --no-cache --config=jest.config.js",
         "snapshots:update": "cross-env BABEL_ENV=jest jest --silent -u --config=jest.config.js",
-        "lint": "npm run eslint && npm run tslint",
-        "lint:fix": "eslint './src/js/**/*.js' --fix && tslint './src/js/**/*.ts' --project './tsconfig.json' --fix",
-        "eslint": "eslint './src/js/**/*.js'",
+        "lint": "npm run eslint",
+        "lint:fix": "eslint './src/js/**/*.(j,t)s' --fix",
```
Comment (matijs) [inline package.json]:
```suggestion
        "lint:fix": "eslint './src/js/**/*.{j,t}s' --fix",
```

Comment (matijs) [reply package.json]: We actually have `.js` files?

File: package.json [hunk 2]
```diff
@@ -76,12 +76,13 @@
...
-        "eslint": "^4.19.1",
+        "eslint": "^5.16.0",
+        "eslint-config-airbnb-base": "^13.1.0",
         "eslint-friendly-formatter": "^4.0.1",
         "eslint-loader": "^2.1.2",
-        "eslint-plugin-import": "^2.16.0",
-        "eslint-plugin-promise": "3.8.0",
-        "eslint-plugin-vue": "^4.5.0",
+        "eslint-plugin-import": "^2.17.2",
+        "eslint-plugin-promise": "4.1.1",
```
Comment (matijs) [inline package.json]: Does Airbnb need this? We don't seem to use any of these rules ourselves.

File: package.json [hunk 3]
```diff
@@ -76,12 +76,13 @@
         "babel-preset-stage-2": "^6.24.1",
         "babel-preset-vue": "^2.0.2",
         "cross-env": "^5.1.6",
-        "eslint": "^4.19.1",
+        "eslint": "^5.16.0",
+        "eslint-config-airbnb-base": "^13.1.0",
         "eslint-friendly-formatter": "^4.0.1",
```
Comment (matijs) [inline package.json]: Strong opinions on this? Any objections to switching to `codeframe` for example?

File: package.json (hunk 1 above)
Comment (rhelling) [reply package.json]: Yes, stuff we still need to migrate ;-)

File: package.json (hunk 3 above)
Comment (rhelling) [reply package.json]: https://github.com/Ultimaker/Meta/blob/master/languages/javascript_code_conventions.md 🌵

File: .eslintrc.js [hunk 4]
```diff
@@ -14,112 +14,37 @@ module.exports = {
...
-        'plugin:vue/essential'
-    ],
-    // required to lint *.vue files
     plugins: [
-        'vue'
+        '@typescript-eslint',
+    ],
+    extends: [
+        // 'plugin:@typescript-eslint/recommended',
+        'eslint-config-airbnb-base'
```
Comment (dan-entous) [inline .eslintrc.js]: think it would be more ideal to create an Ultimaker.com ruleset. it could be based on airbnb if necessary

Comment (rhelling) [reply .eslintrc.js]: I surely agree on that, but i also hope we can take out some of the disabled rules in the future..

File: package.json (hunk 2 above)
Comment (rhelling) [reply package.json]: I could uninstall the promise plugin, import is still required

**Pull request merged**